src/main/java/web/command_pattern/CommandDistributor.java

```diff
@@ -0,0 +1,38 @@
+package web.command_pattern;
+
+import exception.OrderNotFoundException;
+import web.command_pattern.command.GetOrderByIDCommand;
+import web.command_pattern.command.GetOrderCommand;
+import web.command_pattern.command.PostOrderCommand;
+
+public class CommandDistributor {
+
+    private CommandDistributor(){}
+
+        private static class Singleton{
+            private static final CommandDistributor INSTANCE = new CommandDistributor();
+        }
+
+    public static CommandDistributor getInstance(){
+        return Singleton.INSTANCE;
+    }
+
+    public Command getCommand(String uri, String method) throws OrderNotFoundException {
+
+        if (uri.split("/").length == 3){
+            uri = "/order/{id}";
+        }
```

Comment:
What does this logic do? Do you make any request with uri containing 3 parts the `/order/{id}` request?

Reply:
This is a very straight way to find out if there is an id after the order. if there is, only one command suites (according to the task), of course it's useless in case with more requests we need to handle. Should i find more universal way?

Reply:
I am concerned that the API client may send a request to endpoint `/starships/12` but he or she will receive an order as a response.
By the way, it is preferable to use plural forms for resources representing collections (like the orders in this case).